### PR TITLE
[nrf noup] scripts: west: build: Use sysbuild by default if in NCS dir

### PR DESCRIPTION
**Pull request by nordicjm**
Uses sysbuild by default when building an application which resides in an allowed NCS-based directory when the sysbuild config key is not set

### CODE REVIEW

File: scripts/west_commands/build.py [hunk 1]
```diff
@@ -557,7 +562,34 @@ def _run_cmake(self, board, origin, cmake_opts):
...
+        # Check if this is an ncs-repo directory
+        allow_list = ['zephyr', 'mcuboot', 'sidewalk', 'find-my', 'nrf']
+        in_ncs_repo_dir = False
+
+        for module in zephyr_module.parse_modules(ZEPHYR_BASE, self.manifest):
+            if module.meta['name'] in allow_list:
+                if self.source_dir[0:len(module.project)] == module.project:
+                    path_char = self.source_dir[len(module.project):len(module.project)+1]
+
+                    if path_char == '/' or path_char == '\\':
```
Comment (tejlmand) [inline scripts/west_commands/build.py:574]: Just use the built-in pathlib features, like:
```
Path(source_dir).relative_to(module.project)
```

File: scripts/west_commands/build.py [hunk 2]
```diff
@@ -9,6 +9,11 @@
 import sys
 import yaml
 
+from pathlib import Path
```
Comment (tejlmand) [inline scripts/west_commands/build.py:12]: pathlib is already imported above: https://github.com/nrfconnect/sdk-zephyr/blob/c9eac8821f4cb4554f2ee301472aba704408455b/scripts/west_commands/build.py#L7

**Review by tejlmand — CHANGES_REQUESTED** [review]
A quick glance.

I don't see the need to go though zephyr_module.py, just fetch the manifest projects directly.

File: scripts/west_commands/build.py (hunk 1 above)
Comment (nordicjm) [reply scripts/west_commands/build.py:574]: Updated

File: scripts/west_commands/build.py (hunk 2 above)
Comment (nordicjm) [reply scripts/west_commands/build.py:12]: It does not work without this line

File: scripts/west_commands/build.py [hunk 3]
```diff
@@ -9,6 +9,11 @@
...
+from zephyr_ext_common import ZEPHYR_BASE
+sys.path.append(os.fspath(Path(__file__).parent.parent))
+import zephyr_module
```
[the remark this answers is not in the document]
Comment (nordicjm) [reply scripts/west_commands/build.py:15]: Because this does not work with optional modules

**Comment by nordicjm** [comment]
Has been moved to https://github.com/nrfconnect/sdk-zephyr/pull/1706

File: scripts/west_commands/build.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (tejlmand) [reply scripts/west_commands/build.py:15]: what  does not work ?
Modules passed with `-DEXTRA_ZEPHYR_MODULES` ?
Or inactive Zephyr projects ?

The `manifest.get_projects([])` should return projects if they are active.

**Pull request closed without merging**